### ISSUE TITLE
Add Fullbright Playermodels

### ISSUE THREAD

**Issue by JONERAGILT** [YaLTeR/OpenAG#173]
Already know that you can get models which have pure colors and stuff but was going to suggest adding the ability to make players have no lighting at all, if possible; similar to CPMA and Quake World. So that way they are brighter than the map and kind of stick out a little bit more.
![EUs72CwUwAAvBPC](https://user-images.githubusercontent.com/112528024/196353717-66352231-9240-46f2-a403-1f73532368e9.jpeg)

![6_fb](https://user-images.githubusercontent.com/112528024/196353245-e5f794fa-10d3-43ee-a747-ba2a47442717.png)

**Comment by YaLTeR** [comment]
Would that be legit though?

**Comment by JONERAGILT** [comment]
i think so, it's common ground in quake and cpma; it doesnt affect the game much, just improves visibility

**Comment by JONERAGILT** [comment]
Plus, i think that adding autohop and auto duckroll would be considered more questionable than this, in my opinion g.

**Comment by JONERAGILT** [comment]
If you think about it, just improves visibility; doesnt change the gameplay fundamentally, game is played the same way. autohop and autoduckroll completeley alter way game is played right? Lot's of movement is easier so game is played a lot differently, people don't have to time jumps, it's not like a jump buffer or something.

**Comment by execut4ble** [comment]
Pretty good suggestion. People already use modified models that are pretty much full-bright, so personally I wouldn't say it would breach competitive integrity.

**Comment by ScriptedSnark** [comment]
Do you really think that will be legit? I implemented it two months ago for some tests and it looks like cheat.

Lighting is also part of AG matches.

My old screenshots (dark area for good comparison):

**No fullbright models**
![no_fullbright](https://cdn.discordapp.com/attachments/992746406332866651/1006196374070628462/endcamp0000.png)

**Fullbright models**
![fullbright](https://cdn.discordapp.com/attachments/992746406332866651/1006196374427140096/endcamp0001.png)

**Comment by execut4ble** [comment]
Good point, I didn't consider lighting, I assumed it would just color the textures with a solid color while leaving the shading/lighting effects intact. In that case yes, it's definitely some sort of ESP.

**Comment by JONERAGILT** [comment]
Yes, that is a good point, but i feel on a broader spectrum; it only plays a part in niche scenarios, at least from what i've seen. could it be considered fair that someone can sit in a corner without being spotted and being able to cheese that aspect of the game? i feel like it helps rectify random factors. plus ag is generally a very fast game, there will not be many instances where you would be trying to hide in plain sight. either way, a good point nontheless.

**Comment by ScriptedSnark** [comment]
> plus ag is generally a very fast game, there will not be many instances where you would be trying to hide in plain sight.

Whole AG != crossfire.bsp xDDDD

Look, there are maps like `undertow`, `lost_village2`, `elixir`, `gasworks`, `last_call`, `scary_1`, `merlin`, `lambda_bunker`, `endworld` (there are many cool maps like these) and so on where gameplay isn't fast/dynamic and requires some concentration and tactics (especially `undertow` - [example](https://youtu.be/UomRCNav13U))

Fullbright playermodels will be huge advantage in these cases (don't think that crossfire/ag_crossfire/bootbox are those who only play, `undertow`, `lost_village2` and etc. matter too). I wanted to add that feature in my client but I also listen to community. Those who thought wisely preferred to vote against the feature.

I am against it too. AG is not Quake.

**Comment by sobakaruzhaya** [comment]
banal comparison:
![endcamp0000](https://user-images.githubusercontent.com/90552375/196412887-9f5d5efa-08b7-49b0-b7c5-d9671f2ceedb.png)
![endcamp0001](https://user-images.githubusercontent.com/90552375/196412881-2fbb42a0-6416-4669-9e00-83a6fc1d075f.png)

**Comment by JONERAGILT** [comment]
Alright, i suppose i respect your decision, this is essentially coming from someone with experience in quake; i don't know too much about ag to be honest, but i thought it would be worth making my points. If it does in someway compromise the sanctity of the gameplay loop then sure, but it could be added in as an optional thing potentially? for servers to allow or not allow, so that if a server does allow fullbright models, people can do that if they want. like an admin command or something.

**Comment by execut4ble** [comment]
Potentially in the distant future yes, when (and if) AG gets rulesets, kinda like in QuakeWorld. I too would like an experience closer to other arena FPS, but it would modify the game too much.

**Comment by UnkwUsr** [comment]
I have this silly thing in my (not working anymore) cheat: 
![image](https://github.com/YaLTeR/OpenAG/assets/49063932/f3829c74-e16a-4934-aba0-dcaa1038b97f)

All done with 2 hooks: StudioRenderModel function from StudioModelRenderer struct, and glColor4f function (see https://github.com/UnkwUsr/hlhax/blob/master/src/funcs/visuals/xqz/xqz.cpp). As of now this is straightforward cheat feature, but it can be adapted to not showing players that are behind the walls.
